lib/ruby-debug-ide/xml_printer.rb

```diff
@@ -389,9 +392,19 @@ def inspect_with_allocation_control(slice, memory_limit)
       result = nil
       inspect_thread = DebugThread.start {
         start_alloc_size = ObjectSpace.memsize_of_all
```

Comment:
Overall looks nice. One last question: do you think such checks on ___every___ call will slow down the execution?
The possible solution there is to make these checks only in 10% of calls (like `Math.random > 0.9`) or alike